src/code/fault.c

```diff
@@ -983,97 +1170,120 @@ void Fault_ThreadEntry(void* arg) {
+            FaultDrawer_DrawText(64, 90, "All Pages are displayed.");
+            FaultDrawer_DrawText(64, 100, "       THANK YOU!       ");
+            FaultDrawer_DrawText(64, 110, " You are great debugger!");
             Fault_WaitForInput();
-        } while (!sFaultStructPtr->exitDebugger);
+        } while (!sFaultInstance->exit);
 
-        while (!sFaultStructPtr->exitDebugger) {}
+        while (!sFaultInstance->exit) {}
 
         Fault_ResumeThread(faultedThread);
     }
 }
 
-void Fault_SetFB(void* fb, u16 w, u16 h) {
-    sFaultStructPtr->fb = fb;
+void Fault_SetFrameBuffer(void* fb, u16 w, u16 h) {
+    sFaultInstance->fb = fb;
     FaultDrawer_SetDrawerFB(fb, w, h);
 }
 
 void Fault_Init(void) {
-    sFaultStructPtr = &gFaultStruct;
-    bzero(sFaultStructPtr, sizeof(FaultThreadStruct));
-    FaultDrawer_SetDefault();
+    sFaultInstance = &gFaultMgr;
+    bzero(sFaultInstance, sizeof(FaultMgr));
+    FaultDrawer_Init();
     FaultDrawer_SetInputCallback(Fault_WaitForInput);
-    sFaultStructPtr->exitDebugger = false;
-    sFaultStructPtr->msgId = 0;
-    sFaultStructPtr->faultHandlerEnabled = false;
-    sFaultStructPtr->faultedThread = NULL;
-    sFaultStructPtr->padCallback = Fault_PadCallback;
-    sFaultStructPtr->clients = NULL;
-    sFaultStructPtr->faultActive = false;
-    gFaultStruct.faultHandlerEnabled = true;
-    osCreateMesgQueue(&sFaultStructPtr->queue, &sFaultStructPtr->msg, 1);
+    sFaultInstance->exit = false;
+    sFaultInstance->msgId = 0;
+    sFaultInstance->faultHandlerEnabled = false;
+    sFaultInstance->faultedThread = NULL;
+    sFaultInstance->padCallback = Fault_PadCallback;
+    sFaultInstance->clients = NULL;
+    sFaultInstance->autoScroll = false;
+    gFaultMgr.faultHandlerEnabled = true;
+    osCreateMesgQueue(&sFaultInstance->queue, &sFaultInstance->msg, 1);
     StackCheck_Init(&sFaultThreadInfo, &sFaultStack, sFaultStack + sizeof(sFaultStack), 0, 0x100, "fault");
-    osCreateThread(&sFaultStructPtr->thread, 2, Fault_ThreadEntry, 0, sFaultStack + sizeof(sFaultStack),
+    osCreateThread(&sFaultInstance->thread, 2, Fault_ThreadEntry, 0, sFaultStack + sizeof(sFaultStack),
                    OS_PRIORITY_APPMAX);
-    osStartThread(&sFaultStructPtr->thread);
+    osStartThread(&sFaultInstance->thread);
 }
 
-void Fault_HangupFaultClient(const char* arg0, const char* arg1) {
-    osSyncPrintf("HungUp on Thread %d\n", osGetThreadId(0));
-    osSyncPrintf("%s\n", arg0 != NULL ? arg0 : "(NULL)");
-    osSyncPrintf("%s\n", arg1 != NULL ? arg1 : "(NULL)");
-    FaultDrawer_Printf("HungUp on Thread %d\n", osGetThreadId(0));
-    FaultDrawer_Printf("%s\n", arg0 != NULL ? arg0 : "(NULL)");
-    FaultDrawer_Printf("%s\n", arg1 != NULL ? arg1 : "(NULL)");
+/**
+ * Fault page for Hungup crashes. Displays the thread id and two messages
+ * specified in arguments to `Fault_AddHungupAndCrashImpl`.
+ */
+void Fault_HangupFaultClient(const char* exp1, const char* exp2) {
```

Comment:
this has bothered me for a long time, any reason this is "Hangup" and not "Hungup"

also why `Hungup` and not `HungUp` since I assume this comes from debug strings anyway

(not your change I know)